torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2913,6 +2913,28 @@ def sample_inputs_diag(op_info, device, dtype, requires_grad, **kwargs):
 
     return samples + [vec_sample]
 
+def sample_inputs_diagonal(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, dtype=dtype, device=device, requires_grad=requires_grad, low=None, high=None)
+
+    # 2D Tensors
+    tensors_2d = (
+        make_arg((M, M), low=None, high=None),
+        make_arg((3, 5), low=None, high=None),
+        make_arg((5, 3), low=None, high=None),
+    )
+
+    # 3D Tensors
+    tensors_3d = (
+        make_arg((M, M, M), low=None, high=None),
+    )
+
+    args_2d = ((), (2,), (-2,), (1,), (2,))
```

Comment:
`2` is repeated here.

Reply:
Sorry, should be fixed in the recent commit.